kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/CachingKms.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption;
+
+import java.time.Duration;
+import java.util.concurrent.CompletionStage;
+
+import javax.crypto.SecretKey;
+
+import com.github.benmanes.caffeine.cache.AsyncLoadingCache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+
+import io.kroxylicious.kms.service.DekPair;
+import io.kroxylicious.kms.service.Kms;
+import io.kroxylicious.kms.service.Serde;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Kms implementation that caches results from a delegate where appropriate.
+ * We can cache resolved aliases for some time, and we can cache decrypted EDEKs
+ * indefinitely.
```

Comment:
From the point of view of this class, the expiration time is configurable, so I would avoid saying indefinitely in the comment.

Reply:
Have updated it to talk more about how cacheable the two operations are